src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -104,7 +104,9 @@ public BulkProcessor(
     this.reporter = reporter;
 
     unsentRecords = new ArrayDeque<>(maxBufferedRecords);
-    recordMap = new ConcurrentHashMap<>(maxBufferedRecords);
+    recordsToReportOnError = reporter != null
+        ? new ConcurrentHashMap<>(maxBufferedRecords)
+        : new ConcurrentHashMap<>();
```

Comment:
```suggestion
        : null
```

We can use `Collections.emptyMap()` if we declare the interface above to me `Map` instead of `ConcurrentMap`. But since we are already gating on `null` all the calls to this map, maybe we can go with `null` here to avoid creating a new map with every object when the error reporter is not used.